VS2017ComCmdHere.inf

```diff
@@ -30,11 +30,11 @@ VS2017ComCmdHere.INF
 HKLM,%UDHERE%,DisplayName,,"%VS2017ComCmdHereName%"
 HKLM,%UDHERE%,UninstallString,,"rundll32.exe syssetup.dll,SetupInfObjectInstallAction DefaultUnInstall 132 %17%\VS2017ComCmdHere.inf"
 HKCR,Directory\Background\Shell\VS2017ComCmdHere,,,"%VS2017ComCmdHereAccel%"
-HKCR,Directory\Background\Shell\VS2017ComCmdHere\command,,,"%11%\cmd.exe /k cd ""%V"" && ""C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat"""
+HKCR,Directory\Background\Shell\VS2017ComCmdHere\command,,,"cmd.exe /k call ""C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat"" && pushd ""%V"""
 HKCR,Directory\Shell\VS2017ComCmdHere,,,"%VS2017ComCmdHereAccel%"
-HKCR,Directory\Shell\VS2017ComCmdHere\command,,,"%11%\cmd.exe /k cd ""%1"" && ""C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat"""
+HKCR,Directory\Shell\VS2017ComCmdHere\command,,,"cmd.exe /k call ""C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat"" && pushd ""%V"""
```

Comment:
This used to cd to `%1` and it's now using `%V`, was that intentional?

Same in some of the others

Reply:
I found that the script has an ability to open specific directory with dedicated environment variable %VSCMD_START_DIR%. I fixed the enterprise script with the following: cmd.exe /k set ""VSCMD_START_DIR=%V"" &&  ".... same ....".

Reply:
I found %v in win10's own "command prompt here" so I guess it works better in some edge cases. It is listed here: https://superuser.com/questions/136838/which-special-variables-are-available-when-writing-a-shell-command-for-a-context

Reply:
I like the pushd trick because it will map a temp drive letter for a network drive. Could do that first then set VSCMD_START_DIR=%cd% to stop the script changing it again?

Reply:
I've tried the `<VSCMD_START_DIR=%cd%>` - it's difficult with PUSHD as the directory is not known until the driver letter is mapped, and %cd% gets expanded when the cmd /c is run. Could use `<SETLOCAL EnableDelayedExpansion>` but it changes behaviour for scripts run later. Putting the PUSHD last solves all the problems :-)

Reply:
yeah, I think `pushd` should be more than enough :)

Reply:
It stopped working for me a while ago, and I never bother to work out why. If it works for you now maybe it's just my machine, but please test it before building vs2019 versions based on it. Happy to test the 2019 version when it's ready :-)